src/libponyrt/actor/actor.c

```diff
@@ -757,8 +783,10 @@ PONY_API void pony_become(pony_ctx_t* ctx, pony_actor_t* actor)
 
 PONY_API void pony_poll(pony_ctx_t* ctx)
 {
+  // TODO: this seems like it could allow muted actors to get `ponyint_actor_run`
+  // which shouldn't be allowed. Fixing might require API changes.
   pony_assert(ctx->current != NULL);
-  ponyint_actor_run(ctx, ctx->current, 1);
+  ponyint_actor_run(ctx, ctx->current, true);
```

Comment:
where does this "1" live now?

Reply:
It's now a boolean check to see if we're polling so we reach our batch after a single message is processed. See the following where you asked `what is polling?`:

https://github.com/ponylang/ponyc/pull/3173/files#diff-b166f8416958ecd20981bbb8d58692b4R310

```c
      // if we've reached our batch limit
      // or if we're polling where we want to stop after one app message
      if(app == batch || polling)
        return batch_limit_reached(actor, polling);
```

Reply:
That would result in the "1" value version needing an extra "if" to validate. otoh, it only happens once so it doesn't really matter from a performance perspective.

I found the batch size value of 1 easier to reason about than what you've switched it to. Is it a personal preference thing or is there something I'm not seeing where it makes another part of the code easier to reason about?

Reply:
This was changed because prior to this change, an actor could get marked as `overloaded` if it was being `polled` via `pony_poll` because then it's `app` message processed count would equal it's `batch` size (both would be `1`).

The following PR summary bullet is the one for this change (might need to be reworded to be more clear):

> Change logic related to how overload is set to ensure it is not
set when polling to avoid overload/mute getting set unnecessarily
due to polling using batch size == 1 (previously; now, batch
size is always the default).